jsapp/js/components/header/searchBoxStore.ts

```diff
@@ -30,7 +30,7 @@ class SearchBoxStore extends Reflux.Store {
   };
 
   init() {
-    history.listen(this.onRouteChange.bind(this));
+    setTimeout(() => router.subscribe(this.onRouteChange.bind(this)));
```

Comment:
If we end up with these ugly `setTimeout`s (which are fine hacks IMHO), it would be good to leave a comment somewhere. Adding same comment to each `setTimeout` would ba an overkill, but perhaps a comment in `legacy.tsx` would suffice.

Reply:
Good idea, I'll comment the router variable.